packages/eclwatch/src/ECLArchiveViewer.ts

```diff
@@ -0,0 +1,343 @@
+                    ;
+                rowDiv
+                    .on("mouseenter", () => {
+                        const backgroundColor = context.hoverBackgroundColor();
+                        const textColor = Palette.textColor(backgroundColor);
+                        iconDiv
+                            .style("background-color", backgroundColor)
+                            .style("color", textColor)
+                            ;
+                        labelDiv
+                            .style("background-color", backgroundColor)
+                            .style("color", textColor)
+                            ;
+                    })
+                    .on("mouseleave", () => {
+                        iconDiv
+                            .style("background-color", null)
+                            .style("color", context.fontColor())
+                            ;
+                        labelDiv
+                            .style("background-color", null)
+                            .style("color", context.fontColor())
+                            ;
+                    })
+                    ;
+                if (d.isFolder) {
+                    rowDiv.on("click", function(ev, i, node) {
+                        const folderDepth = context.getRowElementDepth(rowDiv);
+                        let next = this.nextSibling;
+                        const wasClosed = rowDiv.classed("folder-closed");
+                        if (wasClosed) {
+                            rowDiv.classed("folder-closed", false);
+                            rowDiv.classed("folder-open", true);
+                            iconDiv.attr("class", "row-icon " + context.folderIconOpen());
+                        } else {
+                            rowDiv.classed("folder-closed", true);
+                            rowDiv.classed("folder-open", false);
+                            iconDiv.attr("class", "row-icon " + context.folderIconClosed());
+                        }
+                        while (next !== null) {
+                            const nextDepth = Number(context.getRowElementDepth(d3Select(next)));
+                            if (nextDepth > folderDepth) {
+                                next.style.display = wasClosed ? "flex" : "none";
+                                next = next.nextSibling;
+                            } else {
+                                next = null;
+                            }
+                        }
+                    });
+                } else {
+                    rowDiv.on("click", () => {
+                        element.selectAll(".row-label").style("font-weight", "normal");
+                        labelDiv.style("font-weight", "bold");
+                        const ext = d.label.split(".").pop().toLowerCase();
+                        context.rowClick(ext === "json" ? JSON.stringify(JSON.parse(d.content), null, 4) : d.content);
```

Comment:
I don't think the ArchiveList should be responsible for formatting the content (in fact it shouldn't really known about the content at all)...

Reply:
This is just a convenience feature that allows JSON to be formatted for easy readability. Maybe `codemirror_JSONEditor` should do this by default?